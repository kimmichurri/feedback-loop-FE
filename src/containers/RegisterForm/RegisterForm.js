import React, { Component } from 'react'

export class RegisterForm extends Component {
  constructor() {
    super()
    this.state = {
      full_name: '',
      role: '',
      email: '',
      password_1: '',
      password_2: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({
      [name] : value
    })
  }

  checkPassword = (e) => {
    e.preventDefault()
    const { full_name, role, email, password_1, password_2 } = this.state
    if(password_1 === password_2) {
      const newUser = {
        full_name,
        role,
        email,
        password_1
      }
      this.props.handleLogin(newUser)
    } else {
      console.log('Passwords do not match')
    }
  }

  setRole = (role) => {
    this.setState({
      role
    })
  }

  render() {
    return(
      <form className='login-form'>
        <div className='full-name'>
          <label>Enter Your First and Last Name</label>
          <input 
            type="text" 
            onChange={this.handleChange}  
            name="full_name"
            className="login-input full_name"
          />
        </div>
        <div className='email'>
          <label>Email</label>
          <input 
            type="email" 
            onChange={this.handleChange}  
            name="email"
            className="login-input"
          />
        </div>
        <div className='password'>
          <label>Password</label>
          <input 
            type="password" 
            onChange={this.handleChange}  
            name="password_1"
            className="login-input"
          />
        </div>
        <div className='password'>
          <label>Confirm Password</label>
          <input 
            type="password" 
            onChange={this.handleChange}  
            name="password_2"
            className="login-input"
          />
        </div>
        <div className="role-input">
          <p className="role-input-p">I am:</p>
          <div className="role-choice">
            <input type="radio" onClick={() => {this.setRole('Instructor')}} />
            <p className="role-input-p">An Instructor</p>
          </div>
          <div className="role-choice">
            <input type="radio" onClick={() => {this.setRole('Student')}}/>
            <p className="role-input-p">A Student</p>
          </div>
        </div>
        <button className='login-button' onClick={this.checkPassword}>
          Login
        </button> 
      </form>
    )
  }
}

export default RegisterForm