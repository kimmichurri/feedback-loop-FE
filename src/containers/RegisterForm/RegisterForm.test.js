import React from 'react'
import { RegisterForm } from './RegisterForm'
import { shallow } from 'enzyme'

describe('RegisterForm', () => {
  let wrapper
  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<RegisterForm 
      handleLogin={mockFn}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have a default state', () => {
    const expected = {
      full_name: '',
      role: '',
      email: '',
      password_1: '',
      password_2: ''
    }
    expect(wrapper.state()).toEqual(expected)
  })

  it('should update state on change', () => {
    const mockEvent = { target: { name: 'full_name', value: 'First Last'}}
    wrapper.find('.full_name').simulate('change', mockEvent)
    expect(wrapper.state('full_name')).toBe('First Last')
  })

  it('should call handleLogin when checkPassword is called', () => {
    const preventDefault = { preventDefault: jest.fn() }
    const mockState = {
      full_name: 'First Last',
      role: 'Student',
      email: 'user@gmail.com',
      password_1: 'abc',
      password_2: 'abc'
    }
    wrapper.setState(mockState)
    wrapper.instance().checkPassword(preventDefault)
    expect(mockFn).toHaveBeenCalled()
  })

  it.skip('should not call handleLogin if passwords do not match', () => {
    const preventDefault = { preventDefault: jest.fn() }
    const mockState = {
      full_name: 'First Last',
      role: 'Student',
      email: 'user@gmail.com',
      password_1: 'abc',
      password_2: ''
    }
    wrapper.setState(mockState)
    wrapper.instance().checkPassword(preventDefault)
    expect(mockFn).not.toHaveBeenCalled()
  })
})