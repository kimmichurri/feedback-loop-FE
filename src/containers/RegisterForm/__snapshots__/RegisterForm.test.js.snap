// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should match the snapshot 1`] = `
<form
  className="login-form"
>
  <div
    className="full-name"
  >
    <label>
      Enter Your First and Last Name
    </label>
    <input
      className="login-input full_name"
      name="full_name"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="email"
  >
    <label>
      Email
    </label>
    <input
      className="login-input"
      name="email"
      onChange={[Function]}
      type="email"
    />
  </div>
  <div
    className="password"
  >
    <label>
      Password
    </label>
    <input
      className="login-input"
      name="password_1"
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="password"
  >
    <label>
      Confirm Password
    </label>
    <input
      className="login-input"
      name="password_2"
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="role-input"
  >
    <p
      className="role-input-p"
    >
      I am:
    </p>
    <div
      className="role-choice"
    >
      <input
        className="instructor-select"
        name="radio"
        onClick={[Function]}
        type="radio"
      />
      <p
        className="role-input-p"
      >
        An Instructor
      </p>
    </div>
    <div
      className="role-choice"
    >
      <input
        className="student-select"
        name="radio"
        onClick={[Function]}
        type="radio"
      />
      <p
        className="role-input-p"
      >
        A Student
      </p>
    </div>
  </div>
  <button
    className="login-button"
    onClick={[Function]}
  >
    Register
  </button>
</form>
`;
